src/lib.rs

```diff
@@ -155,8 +159,8 @@ pub fn desugar_book(book: &mut Book, opts: CompileOpts) -> Result<Vec<Warning>,
   if opts.eta {
     ctx.book.eta_reduction();
   }
-  if opts.supercombinators {
-    ctx.book.detach_supercombinators();
+  if opts.lift_combinators {
+    ctx.book.lift_combinators(matches!(opts.linearize_matches, OptLevel::Extra));
```

Comment:
I think that it's better to change this `matches!` to an OptLevel method
`ctx.book.lift_combinators(opts.linearize_matches.is_extra())`